components/script/dom/htmlimageelement.rs

```diff
@@ -432,10 +433,14 @@ impl HTMLImageElement {
             _ => {},
         };
 
-        let map = self.upcast::<Node>()
-                      .following_siblings()
-                      .filter_map(Root::downcast::<HTMLMapElement>)
-                      .find(|n| n.upcast::<Element>().get_string_attribute(&LocalName::from("name")) == last);
+        let parent = match self.upcast::<Node>().GetParentNode() {
+            None => return None,
+            Some(p) => p,
+        };
+
+        let map = parent.children()
+                        .filter_map(Root::downcast::<HTMLMapElement>)
+                        .find(|n| n.upcast::<Element>().get_string_attribute(&LocalName::from("name")) == last);
```

Comment:
> The usemap attribute, if specified, must be a valid hash-name reference to a map element.

The map element can be anywhere in the whole tree, AFAIK, not just in the image's parent.